homeassistant/components/egardia.py

```diff
@@ -0,0 +1,114 @@
+"""
+Interfaces with Egardia/Woonveilig alarm control panel.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/egardia/
+"""
+import logging
+import requests
+import voluptuous as vol
+
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_PORT, CONF_HOST, CONF_PASSWORD, CONF_USERNAME, CONF_NAME,
+    EVENT_HOMEASSISTANT_STOP)
+
+REQUIREMENTS = ['pythonegardia==1.0.36']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_REPORT_SERVER_CODES = 'report_server_codes'
+CONF_REPORT_SERVER_ENABLED = 'report_server_enabled'
+CONF_REPORT_SERVER_PORT = 'report_server_port'
+REPORT_SERVER_CODES_IGNORE = 'ignore'
+CONF_VERSION = 'version'
+
+DEFAULT_NAME = 'Egardia'
+DEFAULT_PORT = 80
+DEFAULT_REPORT_SERVER_ENABLED = False
+DEFAULT_REPORT_SERVER_PORT = 52010
+DEFAULT_VERSION = 'GATE-01'
+DOMAIN = 'egardia'
+EGARDIA_SERVER = 'egardia_server'
+EGARDIA_DEVICE = 'egardiadevice'
+EGARDIA_NAME = 'egardianame'
+EGARDIA_REPORT_SERVER_ENABLED = 'egardia_rs_enabled'
+EGARDIA_REPORT_SERVER_CODES = 'egardia_rs_codes'
+NOTIFICATION_ID = 'egardia_notification'
+NOTIFICATION_TITLE = 'Egardia'
+ATTR_DISCOVER_DEVICES = 'egardia_sensor'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_VERSION, default=DEFAULT_VERSION): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_REPORT_SERVER_CODES): vol.All(cv.ensure_list),
```

Comment:
What should this be a list of? A list of strings? Then add that the the validation.
```py
vol.All(cv.ensure_list, [cv.string])
```

Reply:
well actually this should be a dictionary of lists of strings, like this:
```yaml
report_server_codes:
       arm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
       disarm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
       home: XXXXXXXXXXXXXXXX
       triggered: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
       ignore: XXXXXXXXXXXXXXXX
```

Reply:
Try:
```py
SERVER_CODE_SCHEMA = vol.Schema({
    vol.Optional('arm'): vol.All(cv.ensure_list, [cv.string])
    vol.Optional('disarm'): vol.All(cv.ensure_list, [cv.string])
    vol.Optional('home'): vol.All(cv.ensure_list, [cv.string])
    vol.Optional('triggered'): vol.All(cv.ensure_list, [cv.string])
    vol.Optional('ignore'): vol.All(cv.ensure_list, [cv.string])})

...
vol.Optional(CONF_REPORT_SERVER_CODES): SERVER_CODE_SCHEMA,
```

Reply:
Just pushed a new version. Thanks for the help.